libraries/stdlib/src/kotlin/random/Random.kt

```diff
@@ -321,6 +320,136 @@ public fun Random(seed: Int): Random = XorWowRandom(seed, seed.shr(31))
+ */
+@SinceKotlin("1.3")
+public fun Random.nextUInt(range: UIntRange): UInt = when {
+    range.isEmpty() -> throw IllegalArgumentException("Cannot get random in empty range: $range")
+    range.last < UInt.MAX_VALUE -> nextUInt(range.first, range.last + 1.toUInt())
+    range.first > UInt.MIN_VALUE -> nextUInt(range.first, range.last + 1.toUInt())
+    else -> nextUInt()
+}
+
+/**
+ * Gets the next random [ULong] from the random number generator.
+ *
+ * Generated a Unsigned Long random value uniformly distributed between [ULong.MIN_VALUE] and [ULong.MAX_VALUE] (inclusive)
+ */
+@SinceKotlin("1.3")
+public fun Random.nextULong(): ULong = nextLong().toULong()
+
+/**
+ * Gets the next random [ULong] from the random number generator not greater than the specified [bound].
+ *
+ * Generated an Unsigned Long random value uniformly distributed between `0`(inclusive) and the specified [bound] (exclusive).
+ *
+ * @throws IllegalArgumentException if [bound] is zero
+ */
+@SinceKotlin("1.3")
+public fun Random.nextULong(bound: ULong): ULong = nextULong(0uL, bound)
+
+/**
+ * Gets the next random [ULong] from the random number generator in the specified range
+ *
+ * Generated an Unsigned Long random value uniformly distributed between the specified [origin] (inclusive) and the specified [bound] (exclusive).
+ *
+ * @throws IllegalArgumentException if [origin] is greater than or equal to [bound]
+ */
+@SinceKotlin("1.3")
+public fun Random.nextULong(origin: ULong, bound: ULong): ULong {
+    checkULongRangeBounds(origin, bound)
+
+    val originTransformedToLong = origin.toLong() xor Long.MIN_VALUE
+    val boundTransformedToLong = bound.toLong() xor Long.MIN_VALUE
+
+    val randomValueTransformedBack = nextLong(originTransformedToLong, boundTransformedToLong) xor Long.MIN_VALUE
+    return randomValueTransformedBack.toULong()
+}
+
+/**
+ * Gets the next random [ULong] from the random number generator in the specified [range].
+ *
+ * Generates an Unsigned Long random value uniformly distributed in the specified [range]:
+ * from `range.start` inclusive to `range.endInclusive` inclusive.
+ *
+ * @throws IllegalArgumentException if [range] is empty
+ */
+@SinceKotlin("1.3")
+public fun Random.nextULong(range: ULongRange): ULong = when {
+    range.isEmpty() -> throw IllegalArgumentException("Cannot get random in empty range: $range")
+    range.last < ULong.MAX_VALUE -> nextULong(range.first, range.last + 1.toULong())
+    range.first > ULong.MIN_VALUE -> nextULong(range.first, range.last + 1.toULong())
+    else -> nextULong()
+}
+
+/**
+ * Fills the specified byte [array] with random bytes and returns it.
+ *
+ * @return [array] filled with random bytes.
+ */
+@SinceKotlin("1.3")
+public fun Random.nextUBytes(array: UByteArray) = nextUBytes(array, 0, array.size)
+
+/**
+ * Creates a byte array of the specified [size], filled with random bytes.
+ */
+@SinceKotlin("1.3")
+public fun Random.nextUBytes(size: Int): UByteArray = nextUBytes(UByteArray(size))
```

Comment:
Better to delegate each `nextUBytes` to the corresponding `nextBytes` in order to benefit from the better implementations of more specialized overloads.

Reply:
Done.